datalad_osf/git_remote.py

```diff
@@ -27,9 +27,145 @@
 
 from osfclient import OSF
 
+class GitGutter(object):
+    pass
+
+from datalad.distribution.dataset import require_dataset
+from datalad.dochelpers import borrowkwargs
+from datalad.downloaders.base import BaseDownloader
+from datalad.downloaders.providers import Provider
+from datalad.support.network import URL
+
+
+class GGDownloader(BaseDownloader):
+
+    @borrowkwargs(BaseDownloader)
+    def __init__(self, credential=None, authenticator=None):
+        assert credential is None
+        assert authenticator is None
+        super().__init__()
+
+    def get_status_from_headers(cls, headers):
+        import pdb; pdb.set_trace()
+
+    def get_downloader_session(self, url):
+        import pdb; pdb.set_trace()
+        return  # no session is needed really?
+
+    def download(self, url, path=None, **kwargs):
+        # assert path is None -- path to download to?!
+        # assert not kwargs  -- {'overwrite': False} -- I thought it is done above!?
+        url_ri = URL(url)
+        ds = require_dataset(None)  # only in CWD, to be used within special remote anyways
+        import pdb; pdb.set_trace()
+        gg = GGExporter(ds.repo.dot_git)
+        if url_ri.hostname == 'refs':
+            print(gg.get_formatted_refs())
+        elif url_ri.hostname == 'archive':
+            # both at once only ATM
+            archive, refs = gg.export_mirror('/tmp/TEMP-sync')
+            print(archive)
+        else:
+            raise ValueError(f"Have no clue how to get {url}")
+
+
+# TODO: need to patch DataLad to allow for extending providers with ones from entry points
+Provider.DOWNLOADERS['gg'] = {'class': GGDownloader}
```

Comment:
here is where/what for I wanted to be able to be able to add new downloaders for datalad. PR https://github.com/datalad/datalad/pull/5963